keps/sig-scheduling/4815-dra-partitionable-devices/README.md

```diff
@@ -647,441 +475,175 @@ type Device struct {
-  // overwritten by the ones provided here.
-  //
-  // The maximum number of capacities is 32.
+  // The maximum number counters in a device is 32.
+  // In addition, the maximum number of all counters
+  // in all devices is 1024 (for example, 64 devices with
+  // 16 counters each).
   //
   // +required
-  Capacity map[QualifiedName]DeviceCapacity
+  Counters map[string]Counter
 }
 
-// CapacityPool defines a named pool of capacities
-// that are available to be used by devices defined in the
-// ResourceSlice.
-//
-// The capacities are not allocatable by themselves, but
-// can be referenced by devices. When a device is allocated,
-// the capacity it uses will no longer be available for use
-// by other devices.
-type CapacityPool struct {
-  // Name defines the name of the capacity pool.
-  // It must be a DNS label.
-  //
-  // +required
-  Name string
-
-  // Includes defines the set of capacity pool mixins that this capacity
-  // pool includes.
-  //
-  // The propertes of each included mixin are applied to this capacity pool in
-  // order. Conflicting properties from multiple mixins are taken from the
-  // last mixin listed that contains them. Properties set on the capacity pool will
-  // always override properties from mixins.
-  //
-  // The mixins referenced here must be defined in the same
-  // ResourceSlice.
-  //
-  // The maximum number of mixins that can be included is 8.
-  //
-  // +optional
-  // +listType=atomic
-  Includes []CapacityPoolMixinRef
-
-  // Capacity defines the set of capacities for this capacity pool
-  // The name of each capacity must be unique in that set.
-  //
-  // To ensure this uniqueness, capacities defined by the vendor
-  // must be listed without the driver name as domain prefix in
-  // their name. All others must be listed with their domain prefix.
-  //
-  // Capacities listed here will always take precedence over any included
-  // from a mixin.
-  //
-  // The maximum number of capacities is 32.
-  //
-  // +required
-  Capacity map[QualifiedName]DeviceCapacity
-}
-
-// DeviceMixinRef defines a reference to a device mixin.
-type DeviceMixinRef struct {
-  // Name refers to the name of a device mixin in the pool.
-  //
-  // +required
-  Name string
-}
-
-// CapacityPoolMixinRef defines a reference from a capacity pool
-// to a capacity pool mixin.
-type CapacityPoolMixinRef struct {
-  // Name is the name of a CapacityPoolMixin.
-  //
-  // +required
-  Name string
-}
-
-type DeviceCapacityConsumptionMixinRef struct {
-  // Name is the name of a DeviceCapacityConsumptionMixin.
-  //
-  // +required
-  Name string
-}
-
-// DeviceCapacityConsumption defines a set of capacities that
-// a device will consume from a capacity pool.
-type DeviceCapacityConsumption struct {
-  // CapacityPool defines the capacity pool from which the
-  // capacities defined (either directly or through a mixin)
-  // will be consumed from.
+// Counter describes a quantity associated with a device.
```

Comment:
But not just one particular device as in "device defined in the ResourceSlice", right?

```suggestion
// Counter describes a quantity that multiple devices consume when in use.
```

It was tempting to say "quantity of something" but that just raises more questions than it answers, so I left out "of something".